### ISSUE TITLE
Redis standalone can't use application.properties to change Redis configuration

### ISSUE THREAD

**Issue by UnfetteredCodeMan** [spring-projects/spring-data-redis#2914]
Can't change the Redis startup port by modifying the application.properties's host,port,password when using RedisTemplate on Redis standalone.
![bedced2a45f8e98e73546c1b22f7e3c](https://github.com/spring-projects/spring-data-redis/assets/63155008/60e261e5-efd3-4ee4-afbb-4e5ff091b529)
![image](https://github.com/spring-projects/spring-data-redis/assets/63155008/fa37cc0f-6746-4aa8-a425-e9092d04332e)

But I can connect to the Redis server by way of RedisStandaloneConfiguration
![image](https://github.com/spring-projects/spring-data-redis/assets/63155008/559608d7-86ac-4517-b58d-c81328341076)

So is spring-boot-starter-data-redis with version 3.2.5 unable to configure Redis startup configurations by modifying configuration files such as application.properties?

**Comment by mp911de** [comment]
If you would like us to spend some time helping you to diagnose the problem, please spend some time describing it and, ideally, providing a [minimal yet complete sample](https://stackoverflow.com/help/mcve) that reproduces the problem.
You can share it with us by pushing it to a separate repository on GitHub or by zipping it up and attaching it to this issue.

**Comment by UnfetteredCodeMan** [comment]
> 如果您希望我们花一些时间帮助您诊断问题，请花一些时间描述它，理想情况下，提供一个[最小但完整的示例](https://stackoverflow.com/help/mcve)来重现问题。您可以通过将其推送到 GitHub 上的单独存储库或将其压缩并附加到此问题来与我们共享它。

This is my redis configuration class
```MyConfig .java
@Configuration
class MyConfig {

  @Bean
  LettuceConnectionFactory connectionFactory() {
    return new LettuceConnectionFactory();
  }

  @Bean
  ReactiveRedisTemplate<String, String> ReactiveRedisTemplate(ReactoveRedisConnectionFactory connectionFactory) {
    return new ReactiveRedisTemplate<>(connectionFactory, RedisSerializationContext.string());
  }
}
```
This is my project maven environment
```pom.xml
<parent>
        <groupId>org.springframework.boot</groupId>
        <artifactId>spring-boot-starter-parent</artifactId>
        <version>3.2.5</version>
        <relativePath/> <!-- lookup parent from repository -->
</parent>
 <dependencies>
        <dependency>
            <groupId>org.springframework.boot</groupId>
            <artifactId>spring-boot-starter-data-redis-reactive</artifactId>
        </dependency>
 </dependencies>
```
And I configured the host,password&port in application.properties
```application.properties
spring.data.redis.host=xxx.com
spring.data.redis.port=14777
spring.data.redis.password=xxxxx
```
Then I wrote a test case intending to pass a list data to my Redis remote server
```
    @Qualifier("ReactiveRedisTemplate")
    @Autowired
    private ReactiveRedisOperations<String, String> operations;
    public Mono<Long> addLink(String userId, String url) {
        return operations.opsForList().leftPush(userId, url);
    }
    @Test
    void testA() {
        Mono<Long> mono = addLink("tks1", "http://google.com");
        log.info("mono: {}", mono.block());
    }
```
And then an error.The error message is as follows：

> 1. org.springframework.data.redis.RedisConnectionFailureException: Unable to connect to Redis
> 2. Unable to connect to localhost/<unresolved>:6379
> 3. Connection refused: no further information: localhost/127.0.0.1:6379

But I can connect to the Redis server by way of RedisStandaloneConfiguration:
```MyConfig .java
@Configuration
class MyConfig {

  @Bean
  LettuceConnectionFactory connectionFactory() {
    RedisStandaloneConfiguration serverConfig = new RedisStandaloneConfiguration("xxx.com", 14777);
        serverConfig.setPassword("xxxxx");
        return new LettuceConnectionFactory(serverConfig, clientConfig);
  }

  @Bean
  ReactiveRedisTemplate<String, String> ReactiveRedisTemplate(ReactoveRedisConnectionFactory connectionFactory) {
    return new ReactiveRedisTemplate<>(connectionFactory, RedisSerializationContext.string());
  }
}
```
Then the data was uploaded successfully

> 2024-05-22T02:23:15.760+08:00  INFO 3296 --- [           main] p.l.LearnSpringBootApplicationTest       : mono: 1

So, spring-boot-starter-data-redis with version 3.2.5 unable to configure Redis startup configurations by modifying configuration files such as application.properties?

**Comment by UnfetteredCodeMan** [comment]
Solved, packaging in pom.xml is set to pom

**Comment by josue270193** [comment]
Hello, I'm having the same issue. I did what @UnfetteredCodeMan mentions to solved my problem, using the application.properties doesn't work at all. In my case, I use spring boot 3.3.0 and redisStandalone.

**Comment by kwleungaj** [comment]
Hi, I've encountered the same issue using Spring Boot 3.4.5. I tried both `spring.redis.host` and `spring.data.redis.host` in application.yml, and I also attempted to set up environment variables, but neither approach worked. It was still connecting to localhost:6379. I tried @UnfetteredCodeMan approach to use `RedisStandaloneConfiguration` and solved the problem.
```
@Bean
public RedisConnectionFactory redisConnectionFactory() {
    RedisStandaloneConfiguration serverConfig = new RedisStandaloneConfiguration("redis", 6379);
    return new LettuceConnectionFactory(serverConfig);
}
```
